src/converse-register.js

```diff
@@ -219,20 +224,37 @@
                      * Parameters:
                      *      (Submit Event) ev - Form submission event.
                      */
-                    if (ev && ev.preventDefault) { ev.preventDefault(); }
-                    var $form = $(ev.target),
-                        $domain_input = $form.find('input[name=domain]'),
-                        domain = $domain_input.val();
-                    if (!domain) {
-                        $domain_input.addClass('error');
-                        return;
+                    var cancel_label = '';
+                    if (!_converse.registration_domain) { 
+                        if (ev && ev.preventDefault) { ev.preventDefault(); }
+                        var $form = $(ev.target),
+                            $domain_input = $form.find('input[name=domain]'),
+                            domain = $domain_input.val();
+                        if (!domain) {
+                            $domain_input.addClass('error');
+                            return;
+                        }
+                        cancel_label = __('Cancel');
+                    }
+                    else {
+                        domain = _converse.registration_domain;
+                    }
+
+                    var reg_req = tpl_registration_request({
+                        cancel: cancel_label,
+                        info_message: _converse.__('Requesting a registration form from the XMPP server')
+                    });
+
+                    if (!_converse.registration_domain) {
+                        $form.find('input[type=submit]').hide()
```

Comment:
No @jcbrand, I was reviewing my own changes. I just realised(probably after the much needed sleep) that there is nothing wrong in the logic. The submit button is not just hidden in this case. Its replaced with the spinner template and the cancel button.

Anyways, I'm moving on to changing the function calls to put the common code into a function.